src/lib/objects/adapter.ts

```diff
@@ -14,12 +24,73 @@ export function makeWakuObjectAdapter(adapter: Adapter, wallet: BaseWallet): Wak
 		return adapter.estimateTransaction(wallet, to, token)
 	}
 
-	async function checkBalance(token: Token): Promise<void> {
-		await adapter.checkBalance(address, token)
+	async function _checkBalance(token: Token): Promise<void> {
+		await checkBalance(address, token)
+	}
+
```

Comment:
Why does the object check balances? Shouldn't the app be just checking the balances onload and maybe periodically?

Reply:
It's because the balance is get through polling and with polling there is always a tradeoff that you either make too many polls and waste resources or too few and then it may be a poor user experience.

I think we should check the balance periodically (let's say once in every minute or so) but if there is a transaction done and it can be expected that changes the balance then it would be good to keep to possibility to force a `checkBalance`.